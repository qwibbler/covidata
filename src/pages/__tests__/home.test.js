import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import Home from '../home';

const mockStore = configureMockStore();
const store = mockStore({
  home: {
    total: {
      today_confirmed: 90,
      today_deaths: 89,
      today_new_confirmed: 87,
      date: '2022.02.02',
    },
  },
});

it('renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <Home randOpacity={() => ({ opacity: '0.25' })} />
      </Router>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
