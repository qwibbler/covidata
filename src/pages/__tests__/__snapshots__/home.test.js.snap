// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="home"
>
  <div
    className="world"
  >
    <p>
      Total: 
    </p>
    <div
      className="info bar"
    >
      <div
        className="div-bg"
      />
      <p>
        Population: 
      </p>
      <p>
        7,924,303,275
      </p>
    </div>
    <div
      className="info bar"
    >
      <div
        className="div-bg"
      />
      <p>
        Confirmed: 
      </p>
      <p>
        90
      </p>
    </div>
    <div
      className="info bar"
    >
      <div
        className="div-bg"
      />
      <p>
        Per Population: 
      </p>
      <p>
        0.00%
      </p>
    </div>
    <div
      className="info bar"
    >
      <div
        className="div-bg"
      />
      <p>
        Deaths: 
      </p>
      <p>
        89
      </p>
    </div>
    <div
      className="info bar"
    >
      <div
        className="div-bg"
      />
      <p>
        Per Population: 
      </p>
      <p>
        0.00%
      </p>
    </div>
    <div
      className="info bar"
    >
      <div
        className="div-bg"
      />
      <p>
        New Confirmed: 
      </p>
      <p>
        87
      </p>
    </div>
    <div
      className="info bar"
    >
      <div
        className="div-bg"
      />
      <p>
        Per Population: 
      </p>
      <p>
        0.00%
      </p>
    </div>
  </div>
  <div
    className="home tabs"
  >
    <div
      className="continent tab"
    >
      <div
        className="div-bg"
        style={
          Object {
            "opacity": "0.25",
          }
        }
      />
      <a
        href="#/continent/Africa/AF"
        onClick={[Function]}
      >
        Africa
      </a>
    </div>
    <div
      className="continent tab"
    >
      <div
        className="div-bg"
        style={
          Object {
            "opacity": "0.25",
          }
        }
      />
      <a
        href="#/continent/Antarctica/AN"
        onClick={[Function]}
      >
        Antarctica
      </a>
    </div>
    <div
      className="continent tab"
    >
      <div
        className="div-bg"
        style={
          Object {
            "opacity": "0.25",
          }
        }
      />
      <a
        href="#/continent/Asia/AS"
        onClick={[Function]}
      >
        Asia
      </a>
    </div>
    <div
      className="continent tab"
    >
      <div
        className="div-bg"
        style={
          Object {
            "opacity": "0.25",
          }
        }
      />
      <a
        href="#/continent/Europe/EU"
        onClick={[Function]}
      >
        Europe
      </a>
    </div>
    <div
      className="continent tab"
    >
      <div
        className="div-bg"
        style={
          Object {
            "opacity": "0.25",
          }
        }
      />
      <a
        href="#/continent/North America/NA"
        onClick={[Function]}
      >
        North America
      </a>
    </div>
    <div
      className="continent tab"
    >
      <div
        className="div-bg"
        style={
          Object {
            "opacity": "0.25",
          }
        }
      />
      <a
        href="#/continent/Oceania/OC"
        onClick={[Function]}
      >
        Oceania
      </a>
    </div>
    <div
      className="continent tab"
    >
      <div
        className="div-bg"
        style={
          Object {
            "opacity": "0.25",
          }
        }
      />
      <a
        href="#/continent/South America/SA"
        onClick={[Function]}
      >
        South America
      </a>
    </div>
  </div>
</section>
`;
